src/main/java/com/datalogics/pdf/security/LunaHsmManager.java

```diff
@@ -73,9 +74,11 @@ public boolean hsmLogin(final HsmLoginParameters parms) {
             throw new IllegalArgumentException("Error while logging into the Luna HSM" + e);
         }
 
-        isLoggedIn = slotManager.isLoggedIn();
+        if (slotManager.isLoggedIn()) {
+            state = ConnectionState.CONNECTED;
+        }
         loadKeyStore();
-        return isLoggedIn;
+        return slotManager.isLoggedIn();
     }
```

Comment:
Returning a boolean seems inconsistent, since we now treat the `HsmManager` as having more than one state. We either need to return the `ConnectionState`, or just declare `hsmLogin()` as returning `void`.